### PR TITLE
microsoft auth

### PR DESCRIPTION
**Pull request by kfarr**

### CODE REVIEW

**Comment by kfarr** [comment]
enterprise users on restricted account systems may see this:

image

File: src/editor/api/auth.js [hunk 1]
```diff
@@ -14,8 +21,55 @@ const signIn = async () => {
...
     });
   } catch (error) {
-    console.error(error);
+    // handle specific error for `auth/account-exists-with-different-credential`
+    if (error.code === 'auth/account-exists-with-different-credential') {
+      // handle the error
+      STREET.notify.errorMessage(
+        `Cannot use Google login with your email, try using Microsoft login instead.`
+      );
+      console.log(
```
Comment (rahulkgupta) [inline src/editor/api/auth.js:30]: nit: maybe remove the console.log

File: src/editor/api/auth.js [hunk 2]
```diff
@@ -14,8 +21,55 @@ const signIn = async () => {
...
+  try {
+    const provider = new OAuthProvider('microsoft.com');
+    const { user } = await signInWithPopup(auth, provider);
+    STREET.notify.successMessage(
+      `Successful login with Microsoft authentication.`
+    );
+    // first signIn to ga
+    if (user.metadata.creationTime !== user.metadata.lastSignInTime) return;
+    sendMetric('Auth', 'newAccountCreation');
+    posthog.capture('user_signed_up', {
```
Comment (rahulkgupta) [inline src/editor/api/auth.js:52]: if you want to track posthog, you could add something like `provider: 'microsoft'` as a kv pair

File: src/editor/api/auth.js [hunk 3]
```diff
@@ -14,8 +21,55 @@ const signIn = async () => {
...
+const signInMicrosoft = async () => {
+  try {
+    const provider = new OAuthProvider('microsoft.com');
+    const { user } = await signInWithPopup(auth, provider);
+    STREET.notify.successMessage(
+      `Successful login with Microsoft authentication.`
+    );
+    // first signIn to ga
+    if (user.metadata.creationTime !== user.metadata.lastSignInTime) return;
+    sendMetric('Auth', 'newAccountCreation');
```
Comment (rahulkgupta) [inline src/editor/api/auth.js:51]: probably should stop using `sendMetric`

**Review by rahulkgupta — APPROVED** [review]

**Pull request merged**